documentation/docs/pipelines/abapEnvironment/introduction.md

```diff
@@ -0,0 +1,92 @@
+# ABAP Environment Pipeline
+
+The goal of the ABAP Environment Pipeline is to enable Continuous Integration for the SAP Cloud Platform ABAP Environment, also known as Steampunk.
+In the current state, the pipeline enables you to pull your Software Components to specifc systems and perform ATC checks. The following steps are performed:
+
+* Create an instance of the SAP Cloud Platform ABAP Environment
+* Configure the Communication Arrangement SAP_COM_0510
+* Pull Git repositories / Software Components to the instance
+* Run ATC Checks
+* Delete the SAP Cloud ABAP Environment system
+
+## Configuration
+
+### 1. Jenkins Server
+
+Configure your Jenkins Server according to the [documentation](https://sap.github.io/jenkins-library/guidedtour/)
+
+### 2. Jenkinsfile
+
+Create a file named `Jenkinsfile` in your repository with the following content:
+
+```
+@Library('piper-lib-os') _
+
+abapEnvironmentPipeline script: this
+```
+
+The annotation `@Library('piper-lib-os')` is a reference to the Jenkins Configuration, where you configured the Piper Library as a "Global Pipeline Library". If you want to **avoid breaking changes** we advise you to use a specific release of the Piper Library instead of the default master branch (see [documentation](https://sap.github.io/jenkins-library/customjenkins/#shared-library))
+
+### 3. Manifest for Service Creation
+
+Create a file `manifest.yml`. The pipeline will create a SAP Cloud Platform ABAP Environment System in the beginning (and delete it in the end). This file describes the ABAP instance, which will be created:
+
+```yml
+---
+create-services:
+- name:   "abapEnvironmentPipeline"
+  broker: "abap"
+  plan:   "16_abap_64_db"
+  parameters: "{ \"admin_email\" : \"user@example.com\", \"description\" : \"System for ABAP Pipeline\" }"
+```
+
+Please be aware that creating a SAP Cloud ABAP Environment instance may incur costs.
+
+### 4. Configuration for the Communication
+
+The communication to the ABAP system is done using a Communication Arrangement. The Communication Arrangement is created during the pipeline via the command `cf create-service-key`. The configuration for the command needs to be stored in a JSON file. Create the file `sap_com_0510.json` in the repository with the following content:
+
+```json
+{
+  "scenario_id": "SAP_COM_0510",
+  "type": "basic"
+}
+```
+
+### 5. Configuration for ATC
+
+Create a file `atcConfig.yml` to store the configuration for the ATC run. In this file, you can specify which Packages or Software Components shall be checked. Please have a look at the step documentation for more details. Here is an example of the configuration:
+
+```yml
+atcobjects:
+  softwarecomponent:
+    - name: "/DMO/REPO"
+```
```

Comment:
Will the standard use case be to rather check Software Components, Packages or both? According to the most used cases I would either add packages section or leave it as it is.
Anyway, I think including the full functionality in this example would make sense IMO.

Reply:
The API is implemented in a way that I would advise against using both. Also, it does not provide any advantage to use both. Using Software Components should be the default for the ABAP Environment (while the other is of course possible).

In the end, this is one suggestion how to configure the pipeline. More information can the be found in the step documentation